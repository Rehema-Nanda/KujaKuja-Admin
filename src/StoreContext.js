// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
