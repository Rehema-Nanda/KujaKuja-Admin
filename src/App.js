import React, { PureComponent } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { library } from "@fortawesome/fontawesome-svg-core";
import {
    faArrowRight, faArrowLeft, faArrowUp, faArrowDown, faCaretDown, faCaretRight, faSearch, faFilter,
    faTint, faHeart, faAppleAlt, faHome, faShieldAlt, faChartLine, faThumbsUp, faChevronLeft, faChevronRight,
} from "@fortawesome/free-solid-svg-icons";
import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import { observer } from "mobx-react";

import GlobalHeader from "./components/GlobalHeader";
import GlobalFooter from "./components/GlobalFooter";
import Login from "./components/Login";
import Dashboard from "./components/Dashboard";
import MyData from "./components/MyData";
import MyDataView from "./components/MyDataView";
import UsersList from "./components/users/list";
import UsersEdit from "./components/users/edit";
import FeaturedIdeasList from "./components/featured_ideas/list";
import FeaturedIdeasEdit from "./components/featured_ideas/edit";
import CountriesList from "./components/countries/list";
import CountriesEdit from "./components/countries/edit";
import LocationsList from "./components/locations/list";
import LocationsEdit from "./components/locations/edit";
import ServicePointsList from "./components/service_points/list";
import ServicePointsEdit from "./components/service_points/edit";
import ServiceTypesList from "./components/service_types/list";
import ServiceTypesEdit from "./components/service_types/edit";
import Config from "./components/config/Config";
import DataReviewList from "./components/data_review/list";
import NoPage from "./components/no_page/NoPage";
import TagFilterList from "./components/tag_filters/list";
import TagFilterEdit from "./components/tag_filters/edit";

import { StoreContext } from "./StoreContext";

library.add(
    faArrowRight, faArrowLeft, faArrowUp, faArrowDown, faCaretDown, faCaretRight, faSearch, faFilter,
    faTint, faHeart, faAppleAlt, faHome, faShieldAlt, faChartLine, faThumbsUp, faChevronLeft, faChevronRight,
);

toast.configure();

class App extends PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            surveyTypes: [{ id: "", name: "Survey" }, { id: "binary", name: "Self-Swipe" }],
            booleanOptions: [{ id: true, name: "Yes" }, { id: false, name: "No" }],
        };
    }

    ////////////////////////////////////////
    // lifecycle methods
    ////////////////////////////////////////

    async componentDidUpdate() {
        const { dashboardStore: { apiService: { getToken } } } = this.context;

        if (!window.location.href.includes("login")) {
            await getToken(); // getToken will redirect to the login page if necessary
        }
    }

    render() {
        const { dashboardStore: { apiService: { isAuthenticated, getProfile } } } = this.context;
        const { surveyTypes, booleanOptions } = this.state;

        return (
            <div className="App">
                <BrowserRouter>
                    <div>

                        <Route render={() => {
                            return <GlobalHeader />;
                        }}
                        />

                        <Switch>
                            <Route
                                exact
                                path="/"
                                render={() => {
                                    return <Dashboard />;
                                }}
                            />

                            <Route
                                exact
                                path="/login"
                                render={() => {
                                    return <Login />;
                                }}
                            />

                            <Route
                                exact
                                path="/mydata"
                                render={(props) => {
                                    return <MyData history={props.history} surveyTypes={surveyTypes} booleanOptions={booleanOptions} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/mydata/*"
                                render={(props) => {
                                    return <MyDataView history={props.history} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/responses"
                                render={(props) => {
                                    return <MyData history={props.history} surveyTypes={surveyTypes} booleanOptions={booleanOptions} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/responses/*"
                                render={(props) => {
                                    return <MyDataView history={props.history} location={props.location} />;
                                }}
                            />

                            <Route
                                exact
                                path="/users"
                                render={(props) => {
                                    return <UsersList history={props.history} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/users/:id"
                                render={(props) => {
                                    return <UsersEdit history={props.history} match={props.match} />;
                                }}
                            />

                            <Route
                                exact
                                path="/featured_ideas"
                                render={(props) => {
                                    return <FeaturedIdeasList history={props.history} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/featured_ideas/:id"
                                render={(props) => {
                                    return <FeaturedIdeasEdit match={props.match} history={props.history} />;
                                }}
                            />

                            <Route
                                path="/countries/:id"
                                render={(props) => {
                                    return <CountriesEdit match={props.match} history={props.history} />;
                                }}
                            />
                            <Route
                                path="/countries"
                                render={(props) => {
                                    return <CountriesList history={props.history} location={props.location} />;
                                }}
                            />

                            <Route
                                exact
                                path="/locations"
                                render={(props) => {
                                    return <LocationsList history={props.history} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/locations/:id"
                                render={(props) => {
                                    return <LocationsEdit history={props.history} match={props.match} />;
                                }}
                            />

                            <Route
                                exact
                                path="/service_points"
                                render={(props) => {
                                    return <ServicePointsList history={props.history} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/service_points/:id"
                                render={(props) => {
                                    return <ServicePointsEdit history={props.history} match={props.match} />;
                                }}
                            />

                            <Route
                                exact
                                path="/service_types"
                                render={(props) => {
                                    return <ServiceTypesList history={props.history} location={props.location} />;
                                }}
                            />
                            <Route
                                exact
                                path="/service_types/:id"
                                render={(props) => {
                                    return <ServiceTypesEdit history={props.history} match={props.match} />;
                                }}
                            />

                            {isAuthenticated() && getProfile().is_admin
                            && (
                                <>
                                    <Route exact path="/config" component={Config} />

                                    <Route
                                        exact
                                        path="/data_review"
                                        render={(props) => {
                                            return <DataReviewList history={props.history} surveyTypes={surveyTypes} booleanOptions={booleanOptions} location={props.location} />;
                                        }}
                                    />

                                    <Route
                                        exact
                                        path="/tag_filters"
                                        render={(props) => {
                                            return <TagFilterList history={props.history} />;
                                        }}
                                    />

                                    <Route
                                        exact
                                        path="/tag_filters/:id"
                                        render={(props) => {
                                            return <TagFilterEdit history={props.history} match={props.match} />;
                                        }}
                                    />
                                </>
                            )}
                            <Route render={(props) => {
                                return <NoPage history={props.history} />;
                            }}
                            />
                        </Switch>

                        <Route render={() => {
                            return <GlobalFooter />;
                        }}
                        />

                    </div>
                </BrowserRouter>

            </div>
        );
    }
}

App.contextType = StoreContext;

export default observer(App);
