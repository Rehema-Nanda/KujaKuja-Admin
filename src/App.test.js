import React from "react";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import App from "./App";
import { StoreProvider } from "./StoreContext";

function WrapwithProviders({ children }) {
    return (
        <StoreProvider>
            <MemoryRouter>
                {children}
            </MemoryRouter>
        </StoreProvider>
    );
}

const wrapper = mount(<App />, {
    wrappingComponent: WrapwithProviders,
});

it("renders without crashing", () => {
    expect(wrapper.debug()).toMatchSnapshot();
});
