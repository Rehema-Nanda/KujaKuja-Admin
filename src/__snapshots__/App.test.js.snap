// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<App>
  <div className=\\"App\\">
    <BrowserRouter>
      <Router history={{...}}>
        <div>
          <Route render={[Function: render]}>
            <GlobalHeader>
              <div>
                <Navbar className=\\"global-header\\" style={{...}} expand=\\"lg\\" tag=\\"nav\\">
                  <nav style={{...}} className=\\"global-header navbar navbar-expand-lg\\">
                    <div className=\\"logo-parent-div\\">
                      <NavLink className=\\"kuja-kuja-logo-nav-link\\" to=\\"/\\" tag={[Function: NavLink]} onClick={[Function: bound toggleMenu]}>
                        <NavLink to=\\"/\\" onClick={[Function: bound onClick]} className=\\"kuja-kuja-logo-nav-link nav-link\\" activeClassName=\\"active\\" aria-current=\\"page\\">
                          <Route path=\\"\\\\\\\\/\\" exact={[undefined]} strict={[undefined]} location={[undefined]}>
                            <Link to=\\"/\\" className=\\"kuja-kuja-logo-nav-link nav-link active\\" style={{...}} aria-current=\\"page\\" onClick={[Function: bound onClick]} replace={false}>
                              <a className=\\"kuja-kuja-logo-nav-link nav-link active\\" style={{...}} aria-current=\\"page\\" onClick={[Function]} href=\\"/\\">
                                <img alt=\\"Logo\\" className=\\"customer-logo\\" src={{...}} />
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </NavLink>
                    </div>
                  </nav>
                </Navbar>
                <Modal isOpen={false} toggle={[Function: bound toggleLogoutConfirmationModal]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} />
              </div>
            </GlobalHeader>
          </Route>
          <Switch>
            <Route exact={true} path=\\"/\\" render={[Function: render]} location={{...}} computedMatch={{...}}>
              <Dashboard>
                <div>
                  <section className=\\"dashboard\\">
                    <div className=\\"title-background-parent\\">
                      <div className=\\"title-background-shape\\" />
                    </div>
                    <Container tag=\\"div\\">
                      <div className=\\"container\\">
                        <Row tag=\\"div\\">
                          <div className=\\"row\\">
                            <Col tag=\\"div\\" widths={{...}}>
                              <div className=\\"col\\">
                                <div className=\\"login-bar\\">
                                  <div className=\\"welcome\\">
                                    <p className=\\"small\\">
                                      Welcome back!
                                    </p>
                                  </div>
                                  <div className=\\"last-login\\">
                                    <p className=\\"small\\">
                                      Last login: 
                                    </p>
                                  </div>
                                </div>
                              </div>
                            </Col>
                          </div>
                        </Row>
                        <Row tag=\\"div\\">
                          <div className=\\"row\\">
                            <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                              <div className=\\"col-sm-8\\">
                                <Row tag=\\"div\\">
                                  <div className=\\"row\\">
                                    <Col tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col\\">
                                        <div className=\\"top-container\\">
                                          <h4 className=\\"label\\">
                                            Total Data Points
                                          </h4>
                                          <div className=\\"data-holder\\">
                                            <h2 className=\\"numbers_big\\">
                                              0
                                            </h2>
                                          </div>
                                        </div>
                                      </div>
                                    </Col>
                                    <Col tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col\\">
                                        <div className=\\"top-container\\">
                                          <h4 className=\\"label\\">
                                            Overall Satisfaction
                                          </h4>
                                          <div className=\\"pie\\">
                                            <div className=\\"pie-holder\\">
                                              <Pie data={{...}} options={{...}} height={64} width={64}>
                                                <ChartComponent data={{...}} options={{...}} height={64} width={64} type=\\"pie\\" legend={{...}} redraw={false} datasetKeyProvider={[Function]}>
                                                  <canvas height={64} width={64} id={[undefined]} onClick={[Function]} />
                                                </ChartComponent>
                                              </Pie>
                                            </div>
                                            <div className=\\"pie-legend\\">
                                              <p className=\\"small\\">
                                                <Badge color=\\"#F8382B\\" style={{...}} pill={true} tag=\\"span\\">
                                                  <span style={{...}} className=\\"badge badge-#F8382B badge-pill\\">
                                                     
                                                  </span>
                                                </Badge>
                                                 
                                                NaN% Unsatisfied
                                              </p>
                                              <p className=\\"small\\">
                                                <Badge color=\\"success\\" style={{...}} pill={true} tag=\\"span\\">
                                                  <span style={{...}} className=\\"badge badge-success badge-pill\\">
                                                     
                                                  </span>
                                                </Badge>
                                                 
                                                NaN% Satisfied
                                              </p>
                                            </div>
                                          </div>
                                        </div>
                                      </div>
                                    </Col>
                                  </div>
                                </Row>
                                <Row className=\\"top-details-row\\" tag=\\"div\\">
                                  <div className=\\"top-details-row row\\">
                                    <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col-sm-4\\">
                                        <div>
                                          <h4 className=\\"label\\">
                                            Active Countries
                                          </h4>
                                          <h2 className=\\"numbers_big left\\">
                                            0
                                          </h2>
                                        </div>
                                      </div>
                                    </Col>
                                    <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col-sm-4\\">
                                        <div>
                                          <h4 className=\\"label\\">
                                            Active Locations
                                          </h4>
                                          <h2 className=\\"numbers_big left\\">
                                            0
                                          </h2>
                                        </div>
                                      </div>
                                    </Col>
                                    <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col-sm-4\\">
                                        <div>
                                          <h4 className=\\"label\\">
                                            Active Service Points
                                          </h4>
                                          <h2 className=\\"numbers_big left\\">
                                            0
                                          </h2>
                                        </div>
                                      </div>
                                    </Col>
                                  </div>
                                </Row>
                                <Row tag=\\"div\\">
                                  <div className=\\"row\\">
                                    <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col-sm-4\\">
                                        <div>
                                          <h4 className=\\"label\\">
                                            Service Types
                                          </h4>
                                          <h2 className=\\"numbers_big left\\">
                                            0
                                          </h2>
                                        </div>
                                      </div>
                                    </Col>
                                    <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                      <div className=\\"col-sm-4\\">
                                        <div>
                                          <h4 className=\\"label\\">
                                            Team Members
                                          </h4>
                                          <h2 className=\\"numbers_big left\\">
                                            0
                                          </h2>
                                        </div>
                                      </div>
                                    </Col>
                                  </div>
                                </Row>
                              </div>
                            </Col>
                            <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                              <div className=\\"col-sm-4\\">
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\">
                                    <div className=\\"top-container-idea-feed\\">
                                      <h4 className=\\"label\\">
                                        This Week&#39;s Featured Ideas
                                      </h4>
                                      <ul className=\\"list-unstyled\\">
                                        <li>
                                          <p>
                                            No featured ideas this week
                                          </p>
                                        </li>
                                      </ul>
                                    </div>
                                  </div>
                                </Col>
                              </div>
                            </Col>
                          </div>
                        </Row>
                      </div>
                    </Container>
                    <Loading>
                      <div className=\\"loading-container\\">
                        <div className=\\"loader\\" />
                      </div>
                    </Loading>
                  </section>
                </div>
              </Dashboard>
            </Route>
          </Switch>
          <Route render={[Function: render]}>
            <GlobalFooter>
              <section className=\\"footer\\">
                <div className=\\"footer-details\\">
                  <div className=\\"footer-logo\\">
                    <img alt=\\"Logo\\" src={{...}} />
                  </div>
                  <div className=\\"footer-social\\">
                    <div className=\\"social-instagram\\" onClick={[Function: onClick]} />
                    <div className=\\"social-fb\\" onClick={[Function: onClick]} />
                    <div className=\\"social-twitter\\" onClick={[Function: onClick]} />
                    <div className=\\"social-medium\\" onClick={[Function: onClick]} />
                  </div>
                </div>
              </section>
            </GlobalFooter>
          </Route>
        </div>
      </Router>
    </BrowserRouter>
  </div>
</App>"
`;
